src/OrbitDB.js

```diff
@@ -15,6 +16,20 @@ const OrbitDBAddress = require('./orbit-db-address')
 const createDBManifest = require('./db-manifest')
 const exchangeHeads = require('./exchange-heads')
 const { isDefined, io } = require('./utils')
+const isNode = require('is-node')
+const Storage = require('orbit-db-storage-adapter')
+
+let storage
+if (isNode) {
+  const leveldown = require('leveldown')
+  storage = Storage(leveldown)
+  storage.preCreate = async (directory, options) => {
+    fs.mkdirSync(directory, { recursive: true })
+  }
+} else {
+  const leveljs = require('level-js')
+  storage = Storage(leveljs)
+}
```

Comment:
Would it make sense to move the setup of the storage to `createInstance`? That way, we could add `storage` to the options in there and the user would be able to pass in what now is `leveldown` in the above code. I think that would get us directly to the point where this is customizable for the user (and we provide the default, as we do for identity and access controllers elsewhere in the code). What do you think?